File: Astromaximum/Classes/utils/cells/AmaxPlanetHourCell.m
//
//  AmaxMoonSignCell.m
//  Astromaximum
//
//  Created by admin on 05.12.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

#import "AmaxPlanetHourCell.h"
#import "Astromaximum-Swift.h"

@implementation AmaxPlanetHourCell

- (void)awakeFromNib {
    [super awakeFromNib];
    hourLabel = (UILabel *)[self viewWithTag:3];
    [hourLabel setText:NSLocalizedString(@"hour_of", @"Planet hour event caption")];
}

- (void)configure:(AmaxSummaryItem *)si
{
    [super configure:si];
    AmaxEvent* e = [si mActiveEvent];
    if (e != nil) {
        [timeLabel setText:[e normalizedRangeString]];
        [eventLabel setText:[NSString stringWithFormat:@"%c",
                             getSymbol(TYPE_PLANET, e.mPlanet0)]];
		[self setColorOf:eventLabel byEventMode:e];
    }
    [self updateInfoButtonWith:si];
}

@end

File: Astromaximum/Classes/utils/cells/AmaxPlanetHourCell.swift
//
//  AmaxMoonSignCell.m
//  Astromaximum
//
//  Created by admin on 05.12.12.
//  Copyright (c) 2012 S&W Axis. All rights reserved.
//

import UIKit

class AmaxPlanetHourCell : AmaxTableCell {

    private var hourLabel: UILabel?

    override func awakeFromNib() {
        super.awakeFromNib()
        hourLabel = (self.viewWithTag(3) as! UILabel)
        hourLabel?.text = NSLocalizedString("hour_of", comment: "Planet hour event caption")
    }

    override func configure(_ extRangeMode: Bool, _ summaryMode: Bool) {
        super.configure(extRangeMode, summaryMode)
        if let e = getActiveEvent() {
            if extRangeMode {
                timeLabel?.numberOfLines = 2
                timeLabel?.text =
                    AmaxEvent.long2String(e.date(at: 0), format: AmaxEvent.monthAbbrDayDateFormatter(), h24: false)
                    + "\n"
                    + AmaxEvent.long2String(e.date(at: 1), format: AmaxEvent.monthAbbrDayDateFormatter(), h24: false)
            }
            else {
                timeLabel?.numberOfLines = 1
                timeLabel?.text = e.normalizedRangeString()
            }
            eventLabel?.text = String(format: "%c",
                                      getSymbol(TYPE_PLANET, e.mPlanet0.rawValue))
            setColorOf(label: timeLabel, byEventMode: e)
        }
    }
}
